File: client/src/app/shared/data-models/summary-report.model.ts
import { Injectable } from "@angular/core";

@Injectable({
    providedIn: 'root'
})
export class SummaryReportModel{
    id: string = "";
    organizationId: string = "";
    reportDescription: string = "";
    employeePayments: { employeeName: string, earnings: number, deductions: number, netPay: number }[] = [];
    totalEarnings: number = 0.0;
    totalDeductions: number = 0.0;
    netPayTotal: number = 0.0;
    reportType: string = "";
    payPeriod: string = ""; // July 2024, etc...
    reportGeneratedDate: string = "";
    status: string = "";
}

File: src/main/java/com/hris/HRIS/model/SummaryReportModel.java
package com.hris.HRIS.model;

import com.hris.HRIS.dto.SummaryReportItem;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Getter
@Setter
@ToString

@Document(collection = "summary-report")
public class SummaryReportModel {
    @Id
    private String id;
    private String organizationId;
    private String reportDescription;
    private List<SummaryReportItem> employeePayments;
    private Double totalEarnings;
    private Double totalDeductions;
    private Double netPayTotal;
    private String reportType;
    private String payPeriod; // July 2024, etc...
    private String reportGeneratedDate;
    private String status;
}
